pkg/selector/outputs/bubbletea.go

```diff
@@ -62,30 +76,92 @@ func (m BubbleTeaModel) Init() tea.Cmd {
 func (m BubbleTeaModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	switch msg := msg.(type) {
 	case tea.KeyMsg:
+		// don't listen for input if currently typing into text field
+		if m.tableModel.filterTextInput.Focused() {
```

Comment:
What do you think about moving this table- and sorting-related logic to the respective view files, and then forwarding the key msg to the `Update()` of tableView and sortingView? Similar to the other "in focus" handling you do there.

Reply:
These if statements exist to prevent the keys used to switch between states from causing a state switch while typing in a text field.